### PR TITLE
sqlsmith: add schema-related operations

### LINKED ISSUE #54961: sqlsmith: handle user-defined schemas
sqlsmith only looks for tables in the `public` schema. With the the 20.2 release, it should also look in user-defined schemas.

See https://github.com/cockroachdb/cockroach/blob/991893b41e254f2ea5738e4cf4c5cf712c9a4e1e/pkg/internal/sqlsmith/schema.go#L211-L228

### PR DESCRIPTION
**Pull request by jordanlewis**
User-defined schemas are now supported in sqlsmith. UDSs will be
randomly generated, and new tables will be randomly included in the
available UDSs. Queries will select from any table, including those
inside of UDSs as well.

Closes #54961.

Release note: None

### CODE REVIEW

File: pkg/internal/sqlsmith/schema.go [hunk 1]
```diff
@@ -359,25 +392,10 @@ func (s *Smither) extractIndexes(
 					Direction: dir,
 				})
 			}
-			row := s.db.QueryRow(fmt.Sprintf(`
```
Comment (rafiss) [inline pkg/internal/sqlsmith/schema.go:362]: we don't we need this 19.2 fallback logic any more?

**Review by rafiss — APPROVED** [review]
lgtm!

File: pkg/internal/sqlsmith/schema.go (hunk 1 above)
Comment (jordanlewis) [reply pkg/internal/sqlsmith/schema.go:362]: I don't think so - each roachtest is run on its branch now, I thought.

Comment (rafiss) [reply pkg/internal/sqlsmith/schema.go:362]: ah yes, that was just changed in the beginning of november. so this lgtm

**Pull request merged**